src/swap/useSwapQuote.ts

```diff
@@ -85,10 +170,81 @@ const useSwapQuote = () => {
           : null,
       }
 
-      return updatedExchangeRate
+      const baseApproveTx = createApproveTransaction(updatedField, quote.unvalidatedSwapTransaction)
+      const baseSwapTx = createSwapTransaction(quote.unvalidatedSwapTransaction)
+
+      let maxGasCosts: Array<[TokenBalance, BigNumber]> = []
+      for (const feeCurrency of feeCurrencies) {
+        // TODO: this is Celo specific, make it generic
+        const gasPrice = BigInt((await getGasPrice(feeCurrency.address ?? undefined)).toFixed())
+
+        const approveTx = {
+          ...baseApproveTx,
+          maxFeePerGas: gasPrice,
+          feeCurrency: !feeCurrency.isNative ? (feeCurrency.address as Address) : undefined,
+          // We could estimate this ourselves,
+          // but for now we'll just use the value from the quote
+          gas: BigInt(quote.approveTransaction.gas),
+        }
+
+        const swapTx = {
+          ...baseSwapTx,
+          maxFeePerGas: gasPrice,
+          feeCurrency: !feeCurrency.isNative ? (feeCurrency.address as Address) : undefined,
+        }
+
+        const maxGasCost = getMaxGasCost([approveTx, swapTx])
+        const maxGasCostInDecimal = new BigNumber(maxGasCost.toString()).shiftedBy(
+          -feeCurrency.decimals
+        )
+        maxGasCosts.push([feeCurrency, maxGasCostInDecimal])
```

Comment:
nit: What do you think about doing `maxGasCosts.push({feeCurrency, maxGasCostInDecimal})` instead?
So then, when we use these values we do something like `result.feeCurrency` instead of `result[0]`?

Reply:
Yep much better 👍